### DISCUSSION TITLE
Why in this demo, linux stdout is bind to a no-exist pl011?

### DISCUSSION THREAD

**Discussion by zlaazlaa** [bao-project/bao-hypervisor#198] [General]
In this demo, Linux is bound to a PL011 UART located at 0x1c090000, but the DTS defines only one PL011 UART at 0x1c0a0000. Why does the Linux boot log still appear on the terminal?
https://github.com/bao-project/bao-demos/blob/426a74912f340302e5baf975b63876ea1958703c/demos/linux%2Bfreertos/devicetrees/fvp-a/linux.dts#L94

**Comment by DavidMCerdeira** [comment]
Greetings,

The UART is set at 0x1c0a0000 in the device tree and the Bao config file.
However,  the `stdout-path` in the chosen node is incorrectly setting the uart output as 0x1c090000.
Without me running this to confirm, the outcome should be that there's no earlycon Linux output, however, Linux will eventually initialize the UART and use it.

Please open an issue on the [bao-demos repo](https://github.com/bao-project/bao-demos) for this, as that's where the error occurs.
Additionally, if you can, please open a PR to fix the wrong  `stdout-path` field in the device.
Otherwise, we will do it when we get the chance.

Regards.

**Reply by zlaazlaa** [reply]
Here’s my understanding of your response:

1.  The earlycon is not configured in the selected UART, so there is no earlycon output in the UART.
2. The console is set to a UART generated by Linux itself. Linux identifies the correct UART from the device tree (DTB) and directs ttyAMA0 to this correct UART.
3. The stdout-path is configured to a non-existent UART.

Since both 2 and 3 are essentially setting the same thing, if 3 is incorrect but 2 is correct, the output will still appear on the UART. Is my understanding correct?

**Reply by DavidMCerdeira** [reply]
You are correct.

**Reply by zlaazlaa** [reply]
> You are correct.  你是对的。

👌, I will check other dts and open a PR later.
